docs/guides/git/index.md

```diff
@@ -0,0 +1,473 @@
+
+## Setting-up your local repository
+
+Create a local clone of your Github repository by using one of these commands in your terminal:
+
+```
+    git clone git://github.com/YOUR_GITHUB_USERNAME/moodle.git YOUR_LOCALDIR
+```
+
+```
+    git clone git@github.com:YOUR_GITHUB_USERNAME/moodle.git YOUR_LOCALDIR
+```
+
+These commands do several jobs for you:
+
+- Creates a new folder
+- Initialises a new local Git repository
+- Sets your Github repository as the remote repository called 'origin'
+- Makes a local checkout of the branch 'master'
+
+:::note
+In the first example, the URL used in the command is `git://github.com`. This gives read-only access to the your Github repository, and the `git push origin` command that appears later in this document will not work.
+
+If you want to be able to update your **origin** repository, use the URL that starts with `git@github.com` (the second of the two `git clone` commands given above). This will give you read and write access to your Github repository at github.com.
+:::
+
+## Keeping your public repository up-to-date
+<!-- cspell:ignore forwardable -->
+Your fork at Github is not updated automatically. To keep it synced with the upstream Moodle repository, you have to fetch the recent changes from the official moodle.git repository. To avoid conflicts, it is strongly recommended that you never modify the standard Moodle branches. Never commit directly into **master** and **MOODLE_XXX_STABLE** branches, but instead create topic branches to work on. In Git terminology, the **master** branch and **MOODLE_XXX_STABLE** branches should always be fast-forwardable.
+
+![git-sync-github.png](./_index/git-sync-github.png)
+
+To keep your public repository up-to-date, we will register the remote repository `git://git.moodle.org/moodle.git` under the **upstream** alias. Then we create a script to be run regularly that fetches changes from the upstream repository and pushes them to your public repository. Note that this procedure will not affect your local repository.
+
+To register the upstream remote use the following command in your Moodle installation root folder:
+
+```
+    git remote add upstream git://git.moodle.org/moodle.git
+```
+
+The following commands can be used to keep the your forked Moodle branches at your Github repository synced with Moodle's upstream repository. You may wish to store them in a script so that you can run it every week after the upstream repository is updated.
+
+```
+ #!/bin/bash
+ git fetch upstream
+ for BRANCH in MOODLE_{19..39}_STABLE MOODLE_{310..311}_STABLE MOODLE_{400..401}_STABLE master; do
+     git push origin refs/remotes/upstream/$BRANCH:refs/heads/$BRANCH
+ done
+```
+
+:::caution
+Never commit directly into **master** and **MOODLE_XXX_STABLE** branches
```

Comment:
Instead of bold, using backticks:
```suggestion
Never commit directly into `master` and `MOODLE_XXX_STABLE` branches
```

Reply:
Hmmm, ok. I can follow that convention in other places too. I had originally reserved the backticks for code, but I can see now that it is used widely for code and highlighting.